src/v/kafka/server/handlers/topics/types.h

```diff
@@ -156,4 +158,42 @@ std::vector<kafka::creatable_topic_configs> report_topic_configs(
   const cluster::metadata_cache& metadata_cache,
   const cluster::topic_properties& topic_properties);
 
+// Either parse configuration or return nullopt
+template<typename T>
+std::optional<T>
+get_config_value(const config_map_t& config, std::string_view key) {
+    if (auto it = config.find(key); it != config.end()) {
+        return boost::lexical_cast<T>(it->second);
+    }
+    return std::nullopt;
+}
+
+// Special case for options where Kafka allows -1
+// In redpanda the mapping is following
+//
+// -1 (feature disabled)   =>  tristate.is_disabled() == true;
+// no value                =>  tristate.has_value() == false;
+// value present           =>  tristate.has_value() == true;
+
+template<typename T>
```

Comment:
Unfortunate that this template function had to be moved to the header in order to use it in `validators.h`.